gadget/gadget.go

```diff
@@ -816,6 +817,24 @@ func validateVolume(vol *Volume, knownFsLabelsPerVolume map[string]map[string]bo
 		var start quantity.Offset
 		if s.Offset != nil {
 			start = *s.Offset
+
+			// The GPT partition table will start from sector 2 with length
+			// 16384 bytes
+			if vol.Schema == schemaGPT {
```

Comment:
There is a caveat here as vol.Schema might still be "" here if not explicitly set in the gadget because `setImplicitForVolume()` has not been called yet. So you should apply this for "" too, and please add a TODO saying that implicit data should be set before validating in the future (but no need for you to change that yet as I have some PRs changing that in the fly).